### PR TITLE
Exception logging fix

### PR DESCRIPTION
**Pull request by SergiiTao**
- change default severity to ERROR
- removed logging from exceptions

### CODE REVIEW

File: common/class.Logger.php [hunk 1]
```diff
@@ -182,19 +187,12 @@ public function register()
...
      */
-    public function log($level, $message, $tags, $errorFile = '', $errorLine = 0)
+    public function log($level, $message, $tags = [])
     {
         if ($this->enabled) {
             $this->disable();
             try {
                 if (defined('CONFIG_PATH')) {
-                    $tags = is_array($tags) ? $tags : [$tags];
-                    if (!empty($errorFile)) {
```
Comment (Skyld) [inline common/class.Logger.php]: This is a BC here. Are you sure that nothing is checking for these keys anywhere? (`file`, `line`)

File: common/class.Logger.php [hunk 2]
```diff
@@ -337,9 +335,20 @@ public static function w($message, $tags = array())
...
-		self::singleton()->log(self::ERROR_LEVEL, $message, $tags);
-        
+    private static function addTrace(array $tags = [])
+    {
+        if (!isset($tags[self::CONTEXT_TRACE])) {
+            $trace = defined('DEBUG_BACKTRACE_IGNORE_ARGS')
+                ? debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS)
+                : debug_backtrace(false);
+
+            $tags[self::CONTEXT_TRACE] = array_slice($trace, 2);
```
Comment (Skyld) [inline common/class.Logger.php]: What is the purpose of this slice here? Could you please explain?

File: common/class.Logger.php [hunk 3]
```diff
@@ -365,14 +371,21 @@ public static function f($message, $tags = array()
      * @access public
      * @author Joel Bout, <joel.bout@tudor.lu>
      * @param  Exception $exception
-     * @return mixed
      */
     public function handleException( Exception $exception)
```
Comment (Skyld) [inline common/class.Logger.php]: Please remove extra space

File: common/class.Logger.php [hunk 4]
```diff
@@ -389,9 +402,8 @@ public function handleException( Exception $exception)
      */
     public function handlePHPErrors($errorNumber, $errorString, $errorFile = null, $errorLine = null, $errorContext = array())
```
Comment (Skyld) [inline common/class.Logger.php]: Please fix all the conventional errors in this method

File: common/class.Logger.php [hunk 5]
```diff
@@ -418,15 +430,24 @@ public function handlePHPErrors($errorNumber, $errorString, $errorFile = null, $
...
-        			self::d('Unsuported PHP error type: '.$errorNumber, 'common_Logger');
+        			self::d('Unsupported PHP error type: '.$errorNumber, 'common_Logger');
         			$severity = self::ERROR_LEVEL;
         			break;
         	}
-        	self::singleton()->log($severity, 'php error('.$errorNumber.'): '.$errorString, array('PHPERROR'), $errorFile, $errorLine);
+
+            self::singleton()
+                ->log($severity,
+                    'php error(' . $errorNumber . '): ' . $errorString,
```
Comment (Skyld) [inline common/class.Logger.php]: Please use `sprintf` instead here

File: common/class.Logger.php [hunk 6]
```diff
@@ -21,9 +21,6 @@
  *
```
Comment (Skyld) [inline common/class.Logger.php]: Please consider fixing the PSR-2 issues in this class (all of them)

**Comment by Skyld** [comment]
Please resolve conflicts with develop

File: common/class.Exception.php [hunk 7]
```diff
@@ -1,45 +1,50 @@
...
+ *
+ * This file is part of Generis Object Oriented API.
+ *
+ * @author lionel.lecaque@tudor.lu
+ * @package generis
+ * @license GPLv2  http://www.opensource.org/licenses/gpl-2.0.php
+ */
+class common_Exception extends Exception
+{
+    public function __construct($message = null, $code = 0, Exception $previous = null)
```
Comment (siwane) [inline common/class.Exception.php]: This contructor seems useless now

**Comment by siwane** [comment]
This is a wonderful job

**Pull request merged**